src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -148,6 +129,23 @@ protected override async Task ProduceDiagnosticsAsync(
                         AddIncludedDiagnostics(builder, analysisResult.GetOtherDiagnostics());
                     }
                 }
+
+                async Task<ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>> GetOrComputeDiagnosticAnalysisResultsAsync(ImmutableArray<StateSet> stateSets)
+                {
+                    // If there was a 'ForceAnalyzeProjectAsync' run for this project, we can piggy back off of the
+                    // prior computed/cached results as they will be a superset of the results we want.
+                    //
+                    // Note: the caller will loop over *its* state sets, grabbing from the full set of data we've cached
+                    // for this project, and filtering down further.  So it's ok to return this potentially larger set.
+                    if (this.Owner._projectToForceAnalysisData.TryGetValue(project, out var box))
+                        return box.Value.diagnosticAnalysisResults;
```

Comment:
this maintains the logic that if we ran "Run Code Analysis" and made no other changes since then, that the normal diagnostic requests can use the cached values from it.  But it has to be on the same Project instance exactly.